src/frontend/tsx/App.tsx

```diff
@@ -55,8 +63,12 @@ export function App () {
     <Beforeunload onBeforeunload={teardown}>
       <div className='app-container'>
         <GamePanel />
-        <Chessboard />
-        <Menu />
+        <Chessboard pieceTheme={pieceTheme} boardTheme={boardTheme}/>
+        <Menu
+          pieceTheme={pieceTheme}
+          setPieceTheme={setPieceTheme}
+          setBoardTheme={setBoardTheme}
+        />
```

Comment:
I understand why you're doing this, since every single visual components needs one or both of the themes. However, I really dislike this method of passing info between the different visual components. Especially with `localStorage`, it shouldn't be necessary since we can directly load the themes in each component.

However, my dislike is not a sufficient reason to not use this tool. Furthermore, you're the one actively writing this code, so if you find that this is the easiest way to access/use the info, I'll let you make that executive decision.